metric/internal/global/meter_test.go

```diff
@@ -0,0 +1,286 @@
+			if !once {
+				wg.Done()
+				once = true
+			}
+			select {
+			case <-finish:
+				return
+			default:
+			}
+		}
+	}()
+
+	wg.Wait()
+	mtr.setDelegate(nonrecording.NewNoopMeterProvider())
+	close(finish)
+}
+
+func testSetupAllInstrumentTypes(t *testing.T, m metric.Meter) (syncfloat64.Counter, asyncfloat64.Counter) {
+
+	afcounter, err := m.AsyncFloat64().Counter("test_Async_Counter")
+	require.NoError(t, err)
+	_, err = m.AsyncFloat64().UpDownCounter("test_Async_UpDownCounter")
+	require.NoError(t, err)
+	_, err = m.AsyncFloat64().Gauge("test_Async_Gauge")
+	require.NoError(t, err)
+
+	_, err = m.AsyncInt64().Counter("test_Async_Counter")
+	require.NoError(t, err)
+	_, err = m.AsyncInt64().UpDownCounter("test_Async_UpDownCounter")
+	require.NoError(t, err)
+	_, err = m.AsyncInt64().Gauge("test_Async_Gauge")
+	require.NoError(t, err)
+
+	_ = m.RegisterCallback([]instrument.Asynchronous{afcounter}, func(ctx context.Context) {
+		afcounter.Observe(ctx, 3)
+	})
+
+	sfcounter, err := m.SyncFloat64().Counter("test_Async_Counter")
+	require.NoError(t, err)
+	_, err = m.SyncFloat64().UpDownCounter("test_Async_UpDownCounter")
+	require.NoError(t, err)
+	_, err = m.SyncFloat64().Histogram("test_Async_Histogram")
+	require.NoError(t, err)
+
+	_, err = m.SyncInt64().Counter("test_Async_Counter")
+	require.NoError(t, err)
+	_, err = m.SyncInt64().UpDownCounter("test_Async_UpDownCounter")
+	require.NoError(t, err)
+	_, err = m.SyncInt64().Histogram("test_Async_Histogram")
+	require.NoError(t, err)
+
+	return sfcounter, afcounter
+}
+
+// This is to emulate a read from an exporter
+func testCollect(t *testing.T, m metric.Meter) {
+	if tMeter, ok := m.(*meter); ok {
+		m, ok = tMeter.delegate.Load().(metric.Meter)
+		if !ok {
+			t.Error("meter was not delegated")
+			return
+		}
+	}
+	tMeter, ok := m.(*testMeter)
+	if !ok {
+		t.Error("collect called on non-test Meter")
+		return
+	}
+	tMeter.collect()
+}
+
+func Test_MeterProvider_delegates_calls(t *testing.T) {
+
+	// The global MeterProvider should directly call the underlying MeterProvider
+	// if it is set prior to Meter() being called.
+
+	// globalMeterProvider := otel.GetMeterProvider
+	globalMeterProvider := &meterProvider{}
+
+	mp := &testMeterProvider{}
+
+	// otel.SetMeterProvider(mp)
+	globalMeterProvider.setDelegate(mp)
+
+	require.Equal(t, 0, mp.count)
+
+	meter := globalMeterProvider.Meter("go.opentelemetry.io/otel/metric/internal/global/meter_test")
+
+	ctr, actr := testSetupAllInstrumentTypes(t, meter)
+
+	ctr.Add(context.Background(), 5)
+
+	testCollect(t, meter) // This is a hacky way to emulate a read from an exporter
+
+	// Calls to Meter() after setDelegate() should be executed by the delegate
+	require.IsType(t, &testMeter{}, meter)
+	if tMeter, ok := meter.(*testMeter); ok {
+		require.Equal(t, 3, tMeter.afCount)
+		require.Equal(t, 3, tMeter.aiCount)
+		require.Equal(t, 3, tMeter.sfCount)
+		require.Equal(t, 3, tMeter.siCount)
+		require.Equal(t, 1, len(tMeter.callbacks))
```

Comment:
Should these be asserted?

```suggestion
		assert.Equal(t, 3, tMeter.afCount)
		assert.Equal(t, 3, tMeter.aiCount)
		assert.Equal(t, 3, tMeter.sfCount)
		assert.Equal(t, 3, tMeter.siCount)
		assert.Equal(t, 1, len(tMeter.callbacks))
```

Reply:
We inconsistently use both require and assert.  I can switch it if that is better.